BaragonAgentService/src/main/java/com/hubspot/baragon/agent/BaragonAgentServiceModule.java

```diff
@@ -149,7 +149,8 @@ public BaragonAgentMetadata providesAgentMetadata(BaragonAgentConfiguration conf
     final String baseAgentUri = String.format(config.getBaseUrlTemplate(), hostname, httpPort, appRoot);
     final String agentId = String.format("%s:%s", hostname, httpPort);
 
-    return new BaragonAgentMetadata(baseAgentUri, agentId, domain, BaragonAgentEc2Metadata.fromEnvironment(), config.getExtraAgentData());
+    final Optional<Integer> loadBalancerPort = config.getLoadBalancerPort().or(Optional.<Integer>absent());
+    return new BaragonAgentMetadata(baseAgentUri, agentId, domain, BaragonAgentEc2Metadata.fromEnvironment(), loadBalancerPort, config.getExtraAgentData());
```

Comment:
you can omit the `.or()` when setting `loadBalancerPort` -- it'd be simpler to just replace `loadBalancerPort` with `config.getLoadBalancerPort()`